chainstate/tx-verifier/src/transaction_verifier/input_output_policy/constraints_accumulator.rs

```diff
@@ -33,13 +36,15 @@ use super::IOPolicyError;
 pub struct ConstrainedValueAccumulator {
     unconstrained_value: Amount,
     timelock_constrained: BTreeMap<BlockDistance, Amount>,
+    token_fee_burn_constrained: Amount,
```

Comment:
This is exactly what I was worried about when I said "this is hard". 

This is not correct.

If I want to put this in mathematical terms, the dot-product of this variable and "unconstrained_value" is not zero. Meaning: You can represent one of them with the other by multiplying by a scalar (-1).

The correct solution for this problem should not involve adding any members here. The local variable `total_burned` should accumulate all the burned values. Then, by the end of the function you should subtract `total_burned` from `unconstrained_value`. This way, before the function call and after the function call, the state is valid and only orthogonal values exist in the struct, and this struct will still form a group, guaranteeing the properties we need without too many invasive tests.

And just a reminder: `timelock_constrained` can be made perfect (mathematically speaking) by changing BlockDistance to NonZeroBlockDistance. Currently, the two variables in the struct are not unrelated. Currently we're manually upholding the invariant that `timelock_constrained[0] == 0`, which is why what we're doing so far is valid.